Src/FluentAssertions/Formatting/PredicateLambdaExpressionValueFormatter.cs

```diff
@@ -134,7 +134,7 @@ private sealed class AndOperatorChainExtractor : ExpressionVisitor
 
         public override Expression Visit(Expression node)
         {
-            if (node.NodeType == ExpressionType.AndAlso)
+            if (node?.NodeType == ExpressionType.AndAlso)
```

Comment:
We had some discussion about this line in #2264 (be aware that the discussions are spread across multiple comment sections).

As I recall when diving into `Expressions` back then my conclusion was that one cannot create a `BinaryExpression` with a null `Left` or `Right` child. That should prevent the two recursive calls on line 140 and 141 from calling `Visit` on a null `node`.
The initial call to `Visit()` on line 62 is guarded by the conditional on line 22.

With the proposed change and if my analysis is wrong, instead of throwing an NRE here, we would add a null `node` to `AndChain` which would then lead to an NRE on line 25 when trying to invoke `ToString()` on the `null`.

I would be fine with `node!.NodeType` to silence Qodana.

Reply:
@jnyrup : I really appreciate how thorougly you review pull requests!

I changed it as you suggested.